tests/python/frontend/caffe/test_forward.py

```diff
@@ -763,6 +763,37 @@ def test_forward_TanH():
     _test_tanh(np.random.rand(10).astype(np.float32))
 
 
+#######################################################################
+# Reduction
+# -----------
+
+
+def _test_reduction(data, **kwargs):
+    """ One iteration of Reduction """
+    _test_op(data, L.Reduction, "Reduction", **kwargs)
+
+
+def test_forward_Reduction():
+    """ Reduction """
+    reduction_op = {"SUM": 1, "ASUM": 2, "SUMSQ": 3, "MEAN": 4}
+    _test_reduction(np.random.rand(10).astype(np.float32), operation=reduction_op["SUM"], axis=0)
+    _test_reduction(
+        np.random.rand(10).astype(np.float32), operation=reduction_op["SUM"], axis=0, coeff=0.0
```

Comment:
Let us add more values for `axis` and `coeff` to cover more conditions.

Reply:
this suggestion is included in new commit. some more codes were introduced to handle reductions on non-tail axis and give back the result equivalent to Caffe framework.